url7a45f192ca_mfullen_ping_pong_net_tgz-pJ8-ping_pong_net_client_io_TestClientNonPPNServerJ8/src/ping/pong/net/server/ClientExample.java

```diff
@@ -0,0 +1,302 @@
+import java.net.Socket;
+import java.net.URISyntaxException;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.*;
+import ping.pong.net.client.Client;
+import ping.pong.net.client.io.IoClient;
+import ping.pong.net.connection.config.ConnectionConfigFactory;
+import ping.pong.net.connection.config.ConnectionConfiguration;
+import ping.pong.net.connection.messaging.MessageListener;
+
+/**
+ *
+ * @author mfullen
+ */
+public class ClientExample
+{
+    private static class DefaultTrustManager implements X509TrustManager
+    {
+        @Override
+        public void checkClientTrusted(X509Certificate[] arg0, String arg1)
+                throws CertificateException
+        {
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate[] arg0, String arg1)
+                throws CertificateException
+        {
+        }
+
+        @Override
+        public X509Certificate[] getAcceptedIssuers()
+        {
+            return null;
+        }
+    }
+
+    public static void ssl1() throws IOException,
+                                     InterruptedException,
+                                     NoSuchAlgorithmException,
+                                     KeyManagementException,
+                                     KeyStoreException,
+                                     UnrecoverableKeyException,
+                                     CertificateException,
+                                     URISyntaxException
+    {
+        {
+
+//            KeyStore ks = KeyStore.getInstance("JKS");
+//
+//            char[] pw = "pingpong123".toCharArray();
+//            ks.load(Thread.currentThread().getContextClassLoader().getResourceAsStream(ConnectionConfiguration.DEFAULT_KEY_STORE), pw);
+//            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("SunX509");
+//            keyManagerFactory.init(ks, pw);
+//            SSLContext ctx = SSLContext.getInstance("SSLv3");
+//            ctx.init(keyManagerFactory.getKeyManagers(), null, new SecureRandom());
+////            ctx.init(new KeyManager[0], new TrustManager[]
+////                    {
+////                        new DefaultTrustManager()
+////                    }, new SecureRandom());
+//            SSLContext.setDefault(ctx);
+            // SSLSocketFactory factory = (SSLSocketFactory) ctx.getSocketFactory();
+
+
+            SSLContext ctx = SSLContext.getInstance("SSLv3");
+            ctx.init(null, null, null);
+
+
+            String absoluteKeyPath = new File(Thread.currentThread().getContextClassLoader().getResource(ConnectionConfiguration.DEFAULT_KEY_STORE).toURI()).getAbsolutePath();
+            System.out.println(absoluteKeyPath);
+            System.setProperty("javax.net.ssl.keyStore", absoluteKeyPath);
+            System.setProperty("javax.net.ssl.keyStorePassword", ConnectionConfiguration.DEFAULT_KEY_STORE_PASSWORD);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code contains a hardcoded password. Hardcoded secrets or credentials can allow attackers to bypass authentication methods and perform malicious actions. We recommend revoking access to resources using this credential and storing future credentials in a management service such as [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/).

[Learn more about the use of hardcoded credentials](https://cwe.mitre.org/data/definitions/798.html)